pkg/endpoint/policy.go

```diff
@@ -423,6 +423,9 @@ func (e *Endpoint) regeneratePolicy(owner Owner) (bool, error) {
 func (e *Endpoint) regenerate(owner Owner) error {
 	e.BuildMutex.Lock()
 	defer e.BuildMutex.Unlock()
+	if err := e.l3PolicyValidation(); err != nil {
```

Comment:
The endpoint doesn't have its mutex locked at this point. I still can't understand why moving `l3PolicyValidation` to this line helps.